kubeflow/hub/api/model_registry_client.py

```diff
@@ -0,0 +1,265 @@
+
+        Raises:
+            ImportError: If model-registry is not installed.
+        """
+        try:
+            from model_registry import ModelRegistry
+        except ImportError as e:
+            raise ImportError(
+                "model-registry is not installed. Install it with:\n\n"  # fmt: skip
+                "  pip install 'kubeflow[hub]'\n"
+            ) from e
+
+        is_http = base_url.startswith("http://")
+        if is_secure is None:
+            is_secure = not is_http
+        if port is None:
+            port = 8080 if is_http else 443
+
+        self._registry = ModelRegistry(
+            server_address=base_url,
+            port=port,
+            author=author,  # type: ignore[arg-type]
+            is_secure=is_secure,
+            user_token=user_token,
+            custom_ca=custom_ca,
+        )
+
+    def register_model(
+        self,
+        name: str,
+        uri: str,
+        *,
+        version: str,
+        model_format_name: str | None = None,
+        model_format_version: str | None = None,
+        author: str | None = None,
+        owner: str | None = None,
+        version_description: str | None = None,
+        metadata: Mapping[str, SupportedTypes] | None = None,
+    ) -> RegisteredModel:
+        """Register a model.
+
+        This registers a model in the model registry. The model is not downloaded,
+        and has to be stored prior to registration.
+
+        Most models can be registered using their URI, along with optional
+        connection-specific parameters, `storage_key` and `storage_path` or,
+        simply a `service_account_name`. URI builder utilities are recommended
+        when referring to specialized storage; for example `utils.s3_uri_from`
+        helper when using S3 object storage data connections.
+
+        Args:
+            name: Name of the model.
+            uri: URI of the model.
+
+        Keyword Args:
+            version: Version of the model. Has to be unique.
+            model_format_name: Name of the model format (e.g., "pytorch", "tensorflow", "onnx").
+                               Used by KServe to select the appropriate serving runtime.
+            model_format_version: Version of the model format (e.g., "2.0", "1.15").
+            author: Author of the model. Defaults to the client author.
+            owner: Owner of the model. Defaults to the client author.
+            version_description: Description of the model version.
+            metadata: Additional version metadata.
+
+        Returns:
+            Registered model.
+        """
+        return self._registry.register_model(
+            name=name,
+            uri=uri,
+            model_format_name=model_format_name,  # type: ignore[arg-type]
+            model_format_version=model_format_version,  # type: ignore[arg-type]
+            version=version,
+            author=author,
+            owner=owner,
+            description=version_description,
```

Comment:
`description` really goes to `ModelVersion` not to `RegisteredModel` so it probably should have be call `version_description` from the beginning.